### ISSUE TITLE
Is there any way to get the URL that is created by using Image component?

### ISSUE THREAD

**Issue by zachrdz** [cloudinary/cloudinary-react#130]
I've been using the Image component with multiple Transformations inside to get the result I'm wanting. Is there a way the grab the URL that the Image component generates when it creates the ```<img src="...."/>``` element? What I'm wanting to do is get the URL as a string to add to the head meta tags of the page as the image to use for SEO and social media preview cards. I'm using Next.js for ssr, so I would be doing this server side. From what I can see, the URL lives in the Image components state, but it is not exposed through the library to use.

**Comment by zachrdz** [comment]
Not sure if there's a better way, but this is what I've come up with so far.

```
import {Image, Transformation} from 'cloudinary-react';
import cloudinary from 'cloudinary-core';
const cloudinaryCore = new cloudinary.Cloudinary({cloud_name: 'my_cloud_name'});

const image = (
    <Image cloudName='my_cloud_name' publicId='folder/some-picture'>
          <Transformation opacity="60"/>
          <Transformation overlay='folder:pic1' height="60"/>
          <Transformation gravity="north_west" opacity="80" x="66" flags="layer_apply"/>
          <Transformation overlay='folder:pic2' height="60"/>
          <Transformation gravity="north_east" opacity="80" x="66" flags="layer_apply"/>
     </Image>
);

const getUrlFromImage = (imgObj) => {
     let url = null;

     if (imgObj) {
          const transArr = [...imgObj.props.children.map(p => new cloudinary.Transformation().fromOptions(p ? p.props : {}))];
          const transStr = transArr.map(t => t.serialize()).join('/');
            
          const genURL = cloudinaryCore.url(imgObj.props.publicId);
          url = `${genURL.split('v1')[0]}${transStr}/v1${genURL.split('v1')[1]}`;
     }

     return url;
}

const srcURL = getUrlFromImage(image);
```

**Comment by shirlymanor** [comment]
@zachrdz, unfortunately, you can't use react to get the full URL. You can use nodeJS for example: `cloudinary.url("sample",{width:300});`

**Comment by zachrdz** [comment]
@shirlymanor Thanks for the response! I'll probably go with the route you've suggested. I was able to get the full url using the example I posted, but it seems the easier way would just be to create the transformation objects manually and then pass them into cloudinary.url method like you've described. Do you know if there's a way to pass in an array of transformations to that .url() method or does it only take a single transformation object as the second parameter? As a workaround I ended up just serializing all the transformations I had and appended it to the url.

**Comment by shirlymanor** [comment]
@zachrdz you can use an array that the result (array.tostring() ) will look like `width:200,opacity:30` Or use rae transformation. For example: cloudinary.url("sample",{raw_transformation:"w_200,h_300,q_auto"})

**Comment by maoznir** [comment]
@zachrdz in the latest version (1.3.0) it is very simple:
You can pass an innerRef param to the Image component and get a ref to the underlying <img> and get the src attribute that way:

let myRef = React.createRef();

//in render()
<Image
  innerRef={myRef}
  cloudName='demo'
  publicId='sample'
/>

**Comment by maoznir** [comment]
Closing as this was solved in version 1.3.0
